app/assets/javascripts/code_listing/user_annotation.ts

```diff
@@ -85,8 +86,12 @@ export class UserAnnotation extends Annotation {
     }
 
     protected get meta(): string {
+        if (!this.permissions.can_see_annotator) {
+            return I18n.t("js.user_annotation.anonymous_message");
+        }
+
         const timestamp = I18n.l("time.formats.annotation", this.createdAt);
-        const user = this.user.name;
+        const user = this.user?.name;
```

Comment:
```suggestion
        const user = this.user!.name;
```
(Using the non null assertion makes more sense semantically to me.)

Reply:
I don't really like the usage of `!`. I see what you mean that it makes more sense since the name should always exist when we get there, but it feels like asking for trouble. 

Wouldn't it be a better option to use `?`, but place a comment next to the line that shows the developer that name should always exist?

Reply:
> Wouldn't it be better to use `?`, but place a comment next to the line that shows the developer that name should always exist?

This seems to me to be exactly the purpose of `!`; to indicate that the developer is certain a value will always be there and won't be null/undefined.